bonds/src/utils/Price.ts

```diff
@@ -1,15 +1,16 @@
 import {
     KLIMA_BCT_PAIR, BCT_USDC_PAIR,
-    KLIMA_MCO2_PAIR, KLIMA_ERC20_V1_CONTRACT
+    KLIMA_MCO2_PAIR
 } from '../../../lib/utils/Constants'
 import { Address, BigDecimal, BigInt, log } from '@graphprotocol/graph-ts'
 import { UniswapV2Pair } from '../../generated/BCTBondV1/UniswapV2Pair'
-import { toDecimal } from '../../../lib/utils/Decimals'
-
-
-let BIG_DECIMAL_1E9 = BigDecimal.fromString('1e9')
-let BIG_DECIMAL_1E12 = BigDecimal.fromString('1e12')
+import { toDecimal, BIG_DECIMAL_1E9, BIG_DECIMAL_1E12 } from '../../../lib/utils/Decimals'
 
+/**
+ * Sekula: TODO: Create a uniform method that would calculate rate
+ * from univ2 pool based on parameters in order not to repeat code.
+ * That method would be referenced from concrete Token classes
+ */
 export function getBCTUSDRate(): BigDecimal {
```

Comment:
We can do something like this:
   ```
 let reservesCall = pair.try_getReserves()
    if (reservesCall.reverted) {
        return BigDecimal.zero()
    }
    let reserves = reservesCall.value
    let reserve0 = reserves.value0.toBigDecimal()
    let reserve1 = reserves.value1.toBigDecimal()

    let bctRate = reserve0.div(reserve1).times(BIG_DECIMAL_1E12)
```
We have to watch out if the returned value is used for div() - it would crash in that case (division by zero)

Reply:
In theory there's always a small amount of LP that is held on by the AMM, so I don't think it would fully go to zero even if someone pulled all liquidity for that specific pair.